dagstore_async.go

```diff
@@ -134,5 +136,32 @@ func (d *DAGStore) initializeShard(ctx context.Context, s *Shard, mnt mount.Moun
 		return
 	}
 
+	// add all cids in the shard to the inverted (cid -> []Shard Keys) index.
+	iterableIdx, ok := idx.(index.IterableIndex)
+	if !ok {
+		_ = d.failShard(s, d.completionCh, "shard index is not iterable")
+		return
+	}
```

Comment:
Do we actually want to fail here? IIUC, the shard was initialized correctly and it's able to operate.